### PR TITLE
DB-6352: integrate spark2.2 changes on branch-2.5

### PR DESCRIPTION
**Pull request by snowspeeder**
Now that Cloudera spark libraries are decoupled from the branch-2.5, it should be able tor run on the installed base of spark 2.2 or 2.1. Following changes picked from spark22-2.5 branch to be compatible on spark-2.2, which should also work on spark-2.1. Initial changes parcel tested on spark2.1 and spark-2.2 cluster. 
Testing done:
- ITs ran on standalone
- on cluster, ran tpch1g on both spark2.1 and spark2.2 installed

### CODE REVIEW

File: hbase_sql/src/main/java/com/splicemachine/stream/index/HTableOutputFormat.java [hunk 1]
```diff
@@ -55,6 +54,9 @@ public Configuration getConf() {
     @Override
     public RecordWriter<byte[],Either<Exception, KVPair>> getRecordWriter(TaskAttemptContext taskAttemptContext) throws IOException, InterruptedException {
         try {
+            if (outputCommitter == null)
+                getOutputCommitter(taskAttemptContext);
+            assert taskAttemptContext != null && taskAttemptContext.getConfiguration() != null:"configuration passed in is null";
```
Comment (yxia92) [inline hbase_sql/src/main/java/com/splicemachine/stream/index/HTableOutputFormat.java]: Shouldn't this assertion be moved before the call of getOutputCommitter()?

**Comment by snowspeeder** [comment]
Right, will move ahead, thanks.

**Comment by snowspeeder** [comment]
Updated the PR with review comments, please take a look. Thanks!

**Comment by yxia92** [comment]
Thanks @snowspeeder !

**Pull request merged**